socket.go

```diff
@@ -258,7 +259,16 @@ connect:
 		return fmt.Errorf("zmq4: got a nil dial-conn to %q", endpoint)
 	}
 
-	zconn, err := Open(conn, sck.sec, sck.typ, sck.id, false, sck.scheduleRmConn)
+	onCloseCb := sck.scheduleRmConn
+	if sck.autoReconnect {
+		onCloseCb = func(c *Conn) {
+			sck.scheduleRmConn(c)
+			time.Sleep(sck.retry)
+			sck.Dial(endpoint)
```

Comment:
calling back into `socket.Dial` will start another `connReaper` goroutine.
are we sure this is ok?